pennylane/pulse/parametrized_hamiltonian.py

```diff
@@ -213,4 +214,23 @@ def __add__(self, H):
 
         return NotImplemented
 
+    def __mul__(self, other):
+        ops = self.ops.copy()
+        coeffs_fixed = self.coeffs_fixed.copy()
+        coeffs_parametrized = self.coeffs_parametrized.copy()
+        if isinstance(other, TensorLike) and qml.math.ndim(other) == 0:
+            coeffs_fixed = [other * c for c in coeffs_fixed]
+            coeffs_parametrized = [
+                lambda p, t, new_c=c: other * new_c(p, t) for c in coeffs_parametrized
+            ]
+            return ParametrizedHamiltonian(coeffs_fixed + coeffs_parametrized, ops)
+        if callable(other):
```

Comment:
What is this used for?

Reply:
To allow users to multiply callables with `ParametrizedHamiltonian`s. Sth like:
```pycon
>>> H = qml.dot(coeffs=[qml.pulse.constant], ops=[qml.PauliX(0)])
>>> new_H = qml.pulse.rect(1, windows=[(1, 3)]) * H
>>> new_H([1], t=3)
1.0*(PauliX(wires=[0]))
>>> new_H([1], t=4)
0.0*(PauliX(wires=[0]))
```

Reply:
Ah, okay! I think we might want to reconsider whether its worth having a type of convenience function that is an envelope (potentially just by having them in a particular file and specifying in the docstring). 

If you multiply `H` with a function that uses parameters, like `np.sin(p[0] * t)`, you need the original function and the function you multiply on to take the same parameters, which feels a bit awkward to me. I played around with it for a few minutes and found the behaviour unintuitive. But it makes lots of sense for an envelope like `qml.pulse.rect`.

Reply:
You are completely right! I will remove the `callable` support in `__mul__` for now.